platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java

```diff
@@ -909,6 +909,52 @@ private void setApiBaseUrl(@NonNull MapboxMapOptions options) {
    */
   @UiThread
   public void setStyleUrl(@NonNull String url) {
+    setStyleUrl(url, null);
+  }
+
+  /**
+   * <p>
+   * Loads a new map style from the specified URL and receive a callback when the style has finished loading.
+   * </p>
+   * {@code url} can take the following forms:
+   * <ul>
+   * <li>{@code Style.*}: load one of the bundled styles in {@link Style}.</li>
+   * <li>{@code mapbox://styles/<user>/<style>}:
+   * retrieves the style from a <a href="https://www.mapbox.com/account/">Mapbox account.</a>
+   * {@code user} is your username. {@code style} is the ID of your custom
+   * style created in <a href="https://www.mapbox.com/studio">Mapbox Studio</a>.</li>
+   * <li>{@code http://...} or {@code https://...}:
+   * retrieves the style over the Internet from any web server.</li>
+   * <li>{@code asset://...}:
+   * reads the style from the APK {@code assets/} directory.
+   * This is used to load a style bundled with your app.</li>
+   * <li>{@code null}: loads the default {@link Style#MAPBOX_STREETS} style.</li>
+   * </ul>
+   * <p>
+   * This method is asynchronous and will return immediately before the style finishes loading.
```

Comment:
This (and the similar comment(s) above) could be worded better—makes it sound like _immediately before_ instead of what actually happens, which is _immediately_ and _before the style finishes loading_.